contents/c++/rvalue.md

```diff
@@ -0,0 +1,169 @@
+# Rvalue references and Move Semantics
+
+Authors: [Tan Jun An](https://github.com/yamidark)
+
+* [Background](#background)
+    * [Value Semantics](#value-semantics)
+    * [Reference Semantics](#reference-semantics)
+* [Move Semantics](#move-semantics)
+    * [rvalue and lvalue references](#rvalue-and-lvalue-references)
+    * [Using rvalue references](#using-rvalue-references)
+    * [Move Semantics](#move-semantics)
+* [rvalue anti-pattern](#rvalue-anti-pattern)
+* [Resources](#resources)
+
+## Background
+*C++* is a general-purpose programming language designed to provide high performance and efficiency for resource-constrained and large systems. The language has since been extended and improved upon, with new standards being released periodically. One such standard, the C++11 standard, improved the performance of the language with features such as *rvalue references* and *Move Semantics*.
+
+In order to better understand the benefits of using Move Semantics, it is important to first understand the two other modes already available in C++, *Value Semantics* and *References Semantics*.
+
```

Comment:
> ... as *rvalue references* and *Move Semantics*.

> ... available in C++, *Value Semantics* and *References Semantics*.

Of the 4 terms mentioned, any reason why the first one is all lower case?

Also, try to use `_italic_` and `**bold**` so that it is easy to see the difference between the two.